### PR TITLE
fence_gce: google-auth instead of oauth2client

### PR DESCRIPTION
**Pull request by helen-fornazier**
oauth2client is deprecated, replacing it by google-auth

### CODE REVIEW

**Comment by fabbione** [comment]
@oalbrigt please let me know what packages are required in CI for this change to be tested properly.

File: fence/agents/gce/fence_gce.py [hunk 1]
```diff
@@ -5,7 +5,8 @@
 sys.path.append("@FENCEAGENTSLIBDIR@")
 
 import googleapiclient
-import oauth2client
+import googleapiclient.discovery
```
Comment (oalbrigt) [inline fence/agents/gce/fence_gce.py]: This is already imported in line 7.

File: fence/agents/gce/fence_gce.py [hunk 2]
```diff
@@ -99,7 +100,7 @@ def main():
 	run_delay(options)
 
 	try:
-		credentials = oauth2client.GoogleCredentials.get_application_default()
+		credentials, _ = google.auth.default()
```
Comment (oalbrigt) [inline fence/agents/gce/fence_gce.py]: This seems to not be available in a lot of distros yet, so I think we should move the import-line from line 9 to this try-statement, and have a fallback to use the old library (except ImportError:).

File: fence/agents/gce/fence_gce.py (hunk 1 above)
Comment (helen-fornazier) [reply fence/agents/gce/fence_gce.py]: importing the package doesn't mean it imported the module and I get an error that googleapiclient.discovery was not found. What I can do is to remove the line 7 and leave just the "import googleapiclient.discovery"

File: fence/agents/gce/fence_gce.py (hunk 2 above)
Comment (helen-fornazier) [reply fence/agents/gce/fence_gce.py]: Ack, I just noticed that googleapiclient already do that if I use the default credentials [1]

I am fixing this PR, sorry for the noise.

[1] https://github.com/google/google-api-python-client/blob/master/googleapiclient/_auth.py#L39

**Comment by helen-fornazier** [comment]
@oalbrigt Hi, could you please take a look in this last version of the PR when you have some time please?

Thanks

**Comment by helen-fornazier** [comment]
I removed the library requirements for google-auth. This requirement should be in googleapiclient, not in fences, so when googleapiclient updates, fences will use google-auth automatically.

googleapiclient is already migrating to google-auth, see:
https://github.com/google/google-api-python-client/pull/493
https://github.com/google/google-api-python-client/pull/499

**Comment by oalbrigt** [comment]
Looks good to me. Does it also work fine with older versions without the credentials part?

**Comment by oalbrigt** [comment]
add to whitelist

**Comment by oalbrigt** [comment]
You'll have to run git rebase upstream/master and git push -f, as I've just merged a PR that restructures the source tree.

**Comment by helen-fornazier** [comment]
> Looks good to me. Does it also work fine with older versions without the credentials part?
add to whitelist

It works from googleapiclient version 1.6.0, how can I add it to the whitelist? It doesn't seem that AC_PYTHON_MODULE accepts a version

**Comment by helen-fornazier** [comment]
> I guess we could just use an except with whatever error it gives in versions < 1.6.0 where it uses the old method before the final exception, as we cant expect all distributions to use the latest versions.

I can't add a try/except there, because the old library only trows a login exception when it tries to use the conn object, not when building it.

I added a verification of the library version and I also added another commit fixing an import regression.
Please check if it is ok now, or if you want modifs, e.g. maybe I could move the check for the version and the import oauth2client.client to the top of the file.

**Comment by helen-fornazier** [comment]
ops, sorry, I just resent fixing coding style, please check this last version of the PR.
Thanks

**Comment by oalbrigt** [comment]
Great. That's much better. You can change the import googleapiclient back to what it was, as it's also required for the version-check now.

**Comment by helen-fornazier** [comment]
The import works like this, I tested it, but if you want me to add the `import googleapiclient`, we will need to also keep the `import googleapiclient.discovery`, what do you prefer?

**Comment by oalbrigt** [comment]
Oh. I didnt expect it to work when you imported sub parts of a library, so we're all good then.

**Comment by krig** [comment]
Hey @helen-fornazier, apologies for breaking the imports!

**Comment by helen-fornazier** [comment]
Hi @krig, no problem.

**Pull request merged**